app/models/pydantic/UserModel.py

```diff
@@ -32,6 +32,17 @@ def check_password(self) -> Self:
                 raise HTTPException(status_code=400, detail=CommonErrorMessages.PASSWORDS_DONT_MATCH)
         return self
 
+class PydanticUserCreate(BaseModel):
+    """
+    This model is meant to be used when we need to create a new user.
+    """
+    login: Login
+    firstname: Name
+    lastname: Name
+    mail: Mail
+    #The admin can choose the password or let it be generated
+    password: Optional[Password] = None
```

Comment:
You did not add a password confirmation.
If you let the user fill it, make sure he does not make typos.